File: dart/com/spoonacular/client/model/inline_response20037.dart
part of openapi.api;

class InlineResponse20037 {
  
  List<InlineResponse2005> meals = [];
  
  InlineResponse20037Nutrients nutrients = null;
  InlineResponse20037();

  @override
  String toString() {
    return 'InlineResponse20037[meals=$meals, nutrients=$nutrients, ]';
  }

  InlineResponse20037.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['meals'] == null) {
      meals = null;
    } else {
      meals = InlineResponse2005.listFromJson(json['meals']);
    }
    if (json['nutrients'] == null) {
      nutrients = null;
    } else {
      nutrients = new InlineResponse20037Nutrients.fromJson(json['nutrients']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (meals != null)
      json['meals'] = meals;
    if (nutrients != null)
      json['nutrients'] = nutrients;
    return json;
  }

  static List<InlineResponse20037> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20037>() : json.map((value) => new InlineResponse20037.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20037> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20037>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20037.fromJson(value));
    }
    return map;
  }
}

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20037.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20037Nutrients
import com.spoonacular.client.model.InlineResponse2005
import scala.collection.immutable.Seq

/**
 * 
 * @param meals 
 * @param nutrients 
 */
case class InlineResponse20037(meals: Seq[InlineResponse2005],
                nutrients: InlineResponse20037Nutrients
                )

object InlineResponse20037 {
    /**
     * Creates the codec for converting InlineResponse20037 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20037] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20037] = deriveEncoder
}
